base/ca/src/main/java/com/netscape/cms/servlet/cert/ListCerts.java

```diff
@@ -316,12 +313,29 @@ public void process(CMSRequest cmsReq) throws EBaseException {
             } catch (Exception e) {
             }
 
+            if (req.getParameter("direction") != null) {
+                direction = req.getParameter("direction").trim();
+                logger.debug("ListCerts: direction: " + direction);
+                switch(direction) {
+                    case "up":
+                        start = Math.max(0, previousStart - maxCount);
+                        break;
+                    case "down":
+                        start = previousStart + previousCount;
+                        break;
+                    case "end":
+                        start = Math.max(0, totalRecordCount - maxCount);
+                        break;
```

Comment:
If the `totalRecordCount` is difficult to obtain accurately with paged results, we might want to remove functionalities that depend on it, e.g. navigation to the end of the list. This will affect the UI.